src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -249,6 +271,8 @@ public Task SendAsync(byte[] data, MessageType type)
             var sendTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
             var message = new SendMessage(data, type, sendTcs);
 
+            _logger.LogDebug("Sending message");
```

Comment:
I don't think so. We have a loop but it is always sending just one message. The only case when you might have more than one iteration will be when you try to send more than one message at the same time and this is OK (and the reason to have the loop)